src/cli_util.rs

```diff
@@ -1001,6 +1004,34 @@ impl WorkspaceCommandHelper {
         Ok(revset.iter().commits(self.repo().store()).try_collect()?)
     }
 
+    /// Resolve a revset any number of revisions (including 0), but require the
+    /// user to indicate if they allow multiple revisions by prefixing the
+    /// expression with `all:`.
+    pub fn resolve_revset_default_single(
+        &self,
+        revision_str: &str,
+        ui: &mut Ui,
+    ) -> Result<Vec<Commit>, CommandError> {
+        if let Some(revision_str) = revision_str.strip_prefix("all:") {
+            self.resolve_revset(revision_str, ui)
```

Comment:
nit: perhaps, `all::x` would be parsed as dag range if we added proper support for parsing `all:` prefix.

Reply:
Good catch! I guess I should do the parsing properly. Let me try that.

Reply:
Actually, I'll leave that for when we've dropped support for `:` so the parse is unambiguous, if you don't mind. I added a TODO here.